File: sdk/python/simple_agent.py
"""
Simple Agent Example - Python SDK

A minimal example showing how to use the Agent SDK.
"""

import asyncio
from claude_agent_sdk import query, ClaudeAgentOptions


async def main():
    """Run a simple agent that explores the codebase"""

    print("Starting simple agent...")

    async for message in query(
        prompt="What files are in this project? Summarize the structure.",
        options=ClaudeAgentOptions(
            allowed_tools=["Read", "Glob", "Grep", "Bash"],
            permission_mode="bypassPermissions",  # Read-only, no confirmation needed
            setting_sources=["project"]  # Load CLAUDE.md and skills
        )
    ):
        if hasattr(message, "result"):
            print(message.result)


if __name__ == "__main__":
    asyncio.run(main())

File: sdk/typescript/simple_agent.ts
/**
 * Simple Agent Example - TypeScript SDK
 *
 * A minimal example showing how to use the Agent SDK.
 */

import { query } from '@anthropic-ai/claude-agent-sdk';

async function main() {
  console.log('Starting simple agent...');

  for await (const message of query({
    prompt: 'What files are in this project? Summarize the structure.',
    options: {
      allowedTools: ['Read', 'Glob', 'Grep', 'Bash'],
      permissionMode: 'bypassPermissions', // Read-only, no confirmation needed
      settingSources: ['project'], // Load CLAUDE.md and skills
    },
  })) {
    if ('result' in message) {
      console.log(message.result);
    }
  }
}

main().catch(console.error);
